src/cloud-api-adaptor/azure/cmd/copy-image/main.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	armcompute "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+)
+
+func main() {
+	var (
+		subscriptionID        string
+		resourceGroup         string
+		location              string
+		userImageName         string
+		managedDiskName       string
+		galleryName           string
+		imageDefinitionName   string
+		imageVersionName      string
+		communityGalleryImage string
+		targetRegionsStr      string
+	)
+
+	flag.StringVar(&subscriptionID, "subscription-id", os.Getenv("AZURE_SUBSCRIPTION_ID"), "Azure subscription ID")
```

Comment:
it should default to the subscription that the user has set in their current az cli configuration